src/main/java/org/sonar/ucfg/UCFGtoProtobuf.java

```diff
@@ -103,19 +103,25 @@ public static UCFG fromProtobufFile(File protobufFile) throws IOException {
   }
 
   private static UCFG deserializeUcfg(FileInputStream fis) throws IOException {
-    Ucfg.UCFG ucfg = Ucfg.UCFG.parseFrom(fis);
-    UCFGBuilder builder = UCFGBuilder.createUCFGForMethod(ucfg.getMethodId()).at(fromProtobuf(ucfg.getLocation()));
-    ucfg.getParametersList().forEach(pId -> builder.addMethodParam(UCFGBuilder.variableWithId(pId)));
-
-    Map<String, UCFGBuilder.BlockBuilder> blockById = ucfg.getBasicBlocksList().stream().collect(Collectors.toMap(Ucfg.BasicBlock::getId, UCFGtoProtobuf::fromProtobuf));
-    for (Map.Entry<String, UCFGBuilder.BlockBuilder> entry : blockById.entrySet()) {
-      if (ucfg.getEntriesList().contains(entry.getKey())) {
-        builder.addStartingBlock(entry.getValue());
-      } else {
-        builder.addBasicBlock(entry.getValue());
+    String methodId = "";
+    try {
+      Ucfg.UCFG ucfg = Ucfg.UCFG.parseFrom(fis);
+      methodId = ucfg.getMethodId();
+      UCFGBuilder builder = UCFGBuilder.createUCFGForMethod(methodId).at(fromProtobuf(ucfg.getLocation()));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'at' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.ucfg%3Asonar-ucfg&pullRequest=7&issues=AWNADNGYS2S-MqecKUfp&open=AWNADNGYS2S-MqecKUfp)

[ISSUE_KEY]: AWNADNGYS2S-MqecKUfp